File: sorting/shell_sort/shell_sort.py
def shell_sort(data):
    n = len(data)
    interval = n // 2
    while interval > 0:
        for i in range(interval, n):
            value = data[i]
            j = i
            while j >= interval and data[j - interval] > value:
                data[j] = data[j - interval]
                j -= interval
            data[j] = value
        interval //= 2

File: sorting/golang/shell_sort.go
package sorting

func ShellSort(data []int) {
	n := len(data)
	interval := n / 2
	for interval > 0 {
		for i := interval; i < n; i++ {
			value := data[i]
			j := i
			for j >= interval && data[j-interval] > value {
				data[j] = data[j-interval]
				j -= interval
			}
			data[j] = value
		}
		interval /= 2
	}
}
